src/avatar.cpp

```diff
@@ -1185,6 +1186,24 @@ faction *avatar::get_faction() const
     return g->faction_manager_ptr->get( faction_your_followers );
 }
 
+bool avatar::is_ally( const Character &p ) const
+{
+    if( p.getID() == getID() ) {
```

Comment:
I think you can unconditionally return `true` here and in the function below. It is not possible to have multiple avatars in a single player game.

Reply:
It can be called from any Character, so any NPC. It might be true now, but if NPCs get the ability to craft (regardless of player interaction), they would piggyback off of the player whenever they are close.

Reply:
avatar is not npc and would never be.

Reply:
The parameter is Character &, so this is valid pseudo code:

```cpp
npc foo = npc()
avatar you = avatar()
you.is_ally(foo)
```

Reply:
Actual code:

```cpp
std::vector<Character *> Character::get_crafting_group() const
{
    return g->get_characters_if( [this]( const Character & guy ) {
        return guy.is_ally( *this )
               && rl_dist( guy.pos(), pos() ) < PICKUP_RANGE
               && get_map().clear_path( pos(), guy.pos(), PICKUP_RANGE, 1, 100 );
    } );
}

std::vector<Character *> game::get_characters_if( const std::function<bool( const Character & )>
        &pred )
{
    std::vector<Character *> result;
    avatar &a = get_avatar();
    if( pred( a ) ) {
        result.push_back( &a );
    }
    for( npc &guy : all_npcs() ) {
        if( pred( guy ) ) {
            result.push_back( &guy );
        }
    }
    return result;
}



```